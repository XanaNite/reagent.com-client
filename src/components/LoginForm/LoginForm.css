.login-form{
    margin-top: 50px;
}

.login-input{
    margin-top: 15px;
}

.login-form input[type=text]{
    width: 80%;
    height: 30px;
}

.login-button{
    text-align: center;
}

.login-form p{
    color: rgb(80, 95, 182);
    font-weight: 500;
    font-size: 15px;
}

.login-button{
    text-align: center;
}

.login-form button{
    background-color: #384CB8;
    border: 0;
    border-radius: 2px;
    width: 75px;
    height: 40px;
    font-size: 16px;
    margin: 25px;
}

.login-form a{
    text-decoration: none;
    color: whitesmoke;
}