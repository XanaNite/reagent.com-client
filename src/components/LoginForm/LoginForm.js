import React from 'react';
import './LoginForm.css';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service';

export default class LoginForm extends React.Component {
    static defaultProps = {
        onLoginSuccess: () => {}
    }
    
    state = { error: null }
    
    handleSubmitJwtAuth = ev =>{
        ev.preventDefault()
    
        this.setState({error: null})
    
        const {agent_email, password} = ev.target
    
        AuthApiService.postLogin({
            agent_email: agent_email.value,
            password: password.value,
        })
        .then(res => {
          if(typeof res.authToken !== 'undefined') {
            localStorage.setItem('user_id', res.id)
            agent_email.value = ''
            password.value = ''
            TokenService.saveAuthToken(res.authToken)
            this.props.onLoginSuccess()
          }
          else {
            this.setState({error: res.error})
          }
        })
        .catch(res => {
            this.setState({error: res.error})
        })
    }

    render(){
        const { error } = this.state
        return(
            <form className="login-form" onSubmit={this.handleSubmitJwtAuth}>
                <div role='alert'>
                    {error && <p className='red'>{error}</p>}
                </div>
                <div className="login-input">
                    <label htmlFor="agent_email">Email address</label>
                    <div>
                        <input type="text" name="agent_email" id="agent_email" />
                    </div>
                </div>
                <div className="login-input">
                    <label htmlFor="password">Password</label>
                    <div>
                        <input type="password" name="password" id="password" />
                    </div>
                </div>
                <div className="login-button">
                    <button type='submit'>Log in</button>
                </div>
            </form>
        )
    }
}