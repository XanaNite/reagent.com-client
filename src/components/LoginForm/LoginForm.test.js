import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';

describe(`LoginForm component`, () => {
    const props = {
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop',
        agent: {
            first_name: "test",
            last_name: "test",
            title: "test",
            city: "test",
            state: "test"
        }
    };

    it('renders the LoginForm given props', () => {
        const wrapper = shallow(<LoginForm {...props} />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
})