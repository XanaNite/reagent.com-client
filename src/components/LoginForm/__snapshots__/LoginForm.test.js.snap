// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the LoginForm given props 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="login-input"
  >
    <label
      htmlFor="agent_email"
    >
      Email address
    </label>
    <div>
      <input
        id="agent_email"
        name="agent_email"
        type="text"
      />
    </div>
  </div>
  <div
    className="login-input"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <div>
      <input
        id="password"
        name="password"
        type="password"
      />
    </div>
  </div>
  <div
    className="login-button"
  >
    <button
      type="submit"
    >
      Log in
    </button>
  </div>
</form>
`;
